core/app/models/spree/order.rb

```diff
@@ -475,6 +475,7 @@ def empty!
       line_items.destroy_all
       adjustments.destroy_all
       shipments.destroy_all
+      promotions.destroy_all
```

Comment:
Can you try `order_promotions.destroy_all` instead? Even if this line only destroys the join table, we want to be clear that we're targetting the join table, not the promotions table.